jax/experimental/maps.py

```diff
@@ -304,7 +304,9 @@ def make_xmap_callable(fun: lu.WrappedFun,
                                     EXPERIMENTAL_SPMD_LOWERING,
                                     *in_avals)
   else:
-    return f.call_wrapped
+    # We have to trace again, because `f` is a linear function, so we can't just return it.
+    final_jaxpr, _, final_consts = pe.trace_to_jaxpr_final(f, mapped_in_avals)
```

Comment:
Correct me if I'm wrong - shouldn't this be using in_avals?  Some tests suggest this is the case.
```python
final_jaxpr, _, final_consts = pe.trace_to_jaxpr_final(f, in_avals)
return core.jaxpr_as_fun(core.ClosedJaxpr(final_jaxpr, final_consts))
```

Reply:
@apaszke - sorry - I'm not sure the correct fix went in with this merge?